TestsFoundation/TestsFoundation/CoreUIElements/Profile/Profile.swift

```diff
@@ -44,7 +44,8 @@ public enum Profile: String, ElementWrapper {
 
     public static func open() {
         // HACK: sometimes a11y stuff doesn't show up...
-        if !Dashboard.profileButton.exists() {
+        let profileButton = Dashboard.profileButton.waitToExist(shouldFail: false)
+        if !profileButton.exists {
```

Comment:
I see that this `exists` property still uses the snapshot != nil check to determine if the Element is there. We could use `rawElement.exists` in Element.exists function as this build-in method was much more reliable than the snapshot method.

Reply:
Yes, that was it. I started the Student E2E test suite on this branch to see if anything went seriously bad.